File: Java/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Deque<Integer> queue = new ArrayDeque<>();

        for (int i = 0; i < progresses.length; i++) {
            queue.offer((int) Math.ceil((100.0 - progresses[i]) / speeds[i]));
        }

        while (!queue.isEmpty()) {
            int current = queue.poll();
            int count = 1;

            while (!queue.isEmpty() && queue.peek() <= current) {
                queue.poll();
                count++;
            }

            answer.add(count);
        }

        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: Python/프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque


def solution(progresses, speeds):
    answer = []

    q = deque()
    for i in range(len(progresses)):
        q.append((100 - progresses[i] + speeds[i] - 1) // speeds[i])

    while q:
        cnt = 1
        cur = q.popleft()
        while q and q[0] <= cur:
            q.popleft()
            cnt += 1
        answer.append(cnt)

    return answer